### PR TITLE
BLOB's default value when convert to not-nullable

### PR DESCRIPTION
**Pull request by beeender**
"" is a char* type takes 1 byte of memory.

### CODE REVIEW

File: realm/realm-library/src/androidTest/java/io/realm/RealmObjectSchemaTests.java [hunk 1]
```diff
@@ -507,6 +506,47 @@ public void setRemoveRequired() {
...
+                    // Skip not-nullable fields .
+                    if (!fieldType.isNullable()) {
+                        break;
+                    }
+                    schema.addField(fieldName, fieldType.getType());
+                    DynamicRealmObject object = realm.createObject(schema.getClassName());
+                    assertTrue(object.isNull(fieldName));
+                    schema.setRequired(fieldName, true);
+                    assertFalse(object.isNull(fieldName));
+                    if (fieldType == FieldType.BLOB) {
```
Comment (kneth) [inline realm/realm-library/src/androidTest/java/io/realm/RealmObjectSchemaTests.java]: funny to see all these `else if` within a `switch` on the same variable 😄

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@
 * [ObjectServer] Fixed a crash when an authentication error happens (#4726).
 * [ObjectServer] Enabled encryption with Sync (#4561).
 * [ObjectServer] Admin users did not connect correctly to the server (#4750).
+* When converting nullable BLOB field to required, `null` values should be converted to `byte[0]` instead of `byte[1]`.
```
Comment (kneth) [inline CHANGELOG.md]: Good catch!

Comment (cmelchior) [inline CHANGELOG.md]: Seems rewriting code from time to time is useful 😛

Comment (cmelchior) [inline CHANGELOG.md]: This should be part of the 3.3.3 changelog not 3.3.2

Comment (beeender) [inline CHANGELOG.md]: very, and I am fixing another one~~ we should do rewriting more often!

**Pull request merged**